grafana/data_source_organization.go

```diff
@@ -0,0 +1,42 @@
+package grafana
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DatasourceOrganization() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana/latest/administration/manage-organizations/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/http_api/org/)
+`,
+		ReadContext: dataSourceOrganizationRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the Organization.",
+			},
+		},
+	}
+}
+
+func dataSourceOrganizationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+	name := d.Get("name").(string)
+	org, err := client.OrgByName(name)
+
+	if err != nil && err.Error() == "404 Not Found" {
+		return nil
+	}
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Couple of things here.

First, how about the following?
```suggestion
	if err != nil {
		if err.Error() == "404 Not Found" {
			return nil
		}

		return diag.FromErr(err)
	}
```

Second, but probably more important: I'm not 100% sure that in the case of a non existing organization the value of `err.Error()` is that string. A quick test shows me that the value is actually `status: 404, body: {"message":"Organization not found"}`.

This clearly raises an issue in the client library, as we should be returning sentinel errors in this case, but as it currently is, this code won't return `nil` when the organization isn't found.